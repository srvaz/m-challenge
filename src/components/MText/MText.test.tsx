import { MText, TEXT_VARIANT } from './index';

import React from 'react';
import { render } from '@testing-library/react';

describe('MText', () => {
  describe('should render correctly', () => {
    test('default', () => {
      const { asFragment } = render(<MText>This is a mock text</MText>)
      expect(asFragment()).toMatchSnapshot();
    });

    test('title', () => {
      const { asFragment } = render(<MText variant={TEXT_VARIANT.TITLE}>This is a mock text</MText>)
      expect(asFragment()).toMatchSnapshot();
    });

    test('subtitle', () => {
      const { asFragment } = render(<MText variant={TEXT_VARIANT.SUBTITLE}>This is a mock text</MText>)
      expect(asFragment()).toMatchSnapshot();
    });

    test('with custom tag', () => {
      const { asFragment } = render(<MText variant={TEXT_VARIANT.SUBTITLE} tag="h2">This is a mock text</MText>)
      expect(asFragment()).toMatchSnapshot();
    });
  });

  test('renders correctly children', () => {
    const mockText = 'This is a mock text';
    const { getByText } = render(<MText>{mockText}</MText>);

    expect(getByText(mockText)).toBeInTheDocument();
  });
});
