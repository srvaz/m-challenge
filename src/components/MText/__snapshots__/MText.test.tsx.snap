// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MText should render correctly default 1`] = `
<DocumentFragment>
  <p
    class="m-text"
  >
    This is a mock text
  </p>
</DocumentFragment>
`;

exports[`MText should render correctly subtitle 1`] = `
<DocumentFragment>
  <p
    class="m-text m-text--subtitle"
  >
    This is a mock text
  </p>
</DocumentFragment>
`;

exports[`MText should render correctly title 1`] = `
<DocumentFragment>
  <p
    class="m-text m-text--title"
  >
    This is a mock text
  </p>
</DocumentFragment>
`;

exports[`MText should render correctly with custom tag 1`] = `
<DocumentFragment>
  <h2
    class="m-text m-text--subtitle"
  >
    This is a mock text
  </h2>
</DocumentFragment>
`;
